salt/states/archive.py

```diff
@@ -63,13 +63,23 @@ def _gen_checksum(path):
 
 
 def _checksum_file_path(path):
-    relpath = '.'.join((os.path.relpath(path, __opts__['cachedir']), 'hash'))
-    if re.match(r'..[/\\]', relpath):
-        # path is a local file
-        relpath = salt.utils.path_join(
-            'local',
-            os.path.splitdrive(path)[-1].lstrip('/\\'),
-        )
+    try:
+        relpath = '.'.join((os.path.relpath(path, __opts__['cachedir']), 'hash'))
+        if re.match(r'..[/\\]', relpath):
+            # path is a local file
+            relpath = salt.utils.path_join(
+                'local',
+                os.path.splitdrive(path)[-1].lstrip('/\\'),
+            )
+    except ValueError as exc:
+        # The path is on a network drive (Windows)
+        if 'path is on drive' in exc.message:
```

Comment:
Checking on the error message 'path is on drive' isn't way to do.  In my case, error message is 'ValueError: path is on mount 'd:', start on mount 'c:''. 
https://github.com/saltstack/salt/issues/47791

Reply:
Cool, thanks